src/simulation_node.py

```diff
@@ -167,38 +141,78 @@ def subscribing_control_topic(self):
         pass
 
 
+    
     def create_robot(self, client_id):
-        """Create the robot based on the dynamics and structure settings."""
-        # Retrieve and parse the configuration for the robot's structure and dynamics
-        for link_name in self.structure_config.keys():
-            # Extract the configuration for this particular link
-            link_params = self.get_parameter(f'structure.{link_name}').get_value()
-            link_dynamics = self.get_parameter(f'dynamics.{link_name}').get_value()
-
-            # Create collision and visual shapes for the link
-            collision_shape_id = p.createCollisionShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_params['dimensions']])
-            visual_shape_id = p.createVisualShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_params['dimensions']])
             
-            # Create the multibody for the link
-            body_id = p.createMultiBody(baseMass=link_params['mass'],
-                                        baseCollisionShapeIndex=collision_shape_id,
-                                        baseVisualShapeIndex=visual_shape_id,
-                                        basePosition=link_params.get('position', [0, 0, 0]),  
-                                        baseInertialFramePosition=link_params.get('inertia_position', [0, 0, 0]),
-                                        physicsClientId=client_id)
-            
-            # Set the dynamics properties for the link
-            p.changeDynamics(body_id, -1,
-                            restitution=link_dynamics['restitution'],
-                            lateralFriction=link_dynamics['lateralFriction'],
-                            spinningFriction=link_dynamics['spinningFriction'],
-                            rollingFriction=link_dynamics['rollingFriction'],
-                            contactDamping=link_dynamics['contactDamping'],
-                            contactStiffness=link_dynamics['contactStiffness'],
-                            physicsClientId=client_id)
-            
-            # Store the body ID for later use (e.g., control or simulation)
-            self.robots.append(body_id)
+        """Create the robot using dynamics and structure settings with joints, using a mesh for the pedestal."""
+        p.loadURDF("plane.urdf", physicsClientId=client_id)
+
+        # Create the collision and visual shapes for world_box and pedestal
+        shapes = {}
+        for link_name, link_config in self.structure_config.items():
```

Comment:
remove the for loop since we only have base now. just directly setting up base will be more straightforward.